### ISSUE TITLE
refetchEvents triggers scroll reset in month view

### ISSUE THREAD

**Issue by xi** [fullcalendar/fullcalendar#6634]
Using fullcalendar 5.10.1 I experienced an issue which sounds a lot like #3153 (which was fixed in 2.8.0), so this might be a regression.

### Reduced Test Case

I adapted the old test case to the current version of fullcalendar: https://jsfiddle.net/q63vg8oa/

### Bug Description

- steps to reproduce
  - open the dayGridMonth View
  - add several events to the same day (so that the row needs to expand)
  - scroll down
  - call `refetchEvents()`
- expected result
  - the scroll position remains the same
- what happens instead?
  - scroll position is moved up

I guess this happens because the events are temporarily removed, so the row gets smaller and the scroll position needs to adjust. When the events are added again the row expands again, but the scroll position remains the same.

**Comment by mamadoo** [comment]
I have same problem and looking for solution. @arshaw

**Comment by nkosarek-zymergen** [comment]
I'm also running into this issue. Seems like it was introduced between v5.6.0 and v5.7.0

**Comment by jslq** [comment]
@nkosarek-zymergen your suggestion was very helpful

**Comment by christophs78** [comment]
Having this issue too.

**Comment by sseemmiin** [comment]
same as me

**Comment by Sir-Gil-Galad** [comment]
me too

**Comment by Sir-Gil-Galad** [comment]
@acerix could you share any updates regarding the issue? is it going to be fixed ?

**Comment by Gatix** [comment]
I'm getting this bug with the resource timeline view at v5.11.0

**Comment by telcy** [comment]
Having same issue. Would be great to have it fixed.

**Comment by rloview** [comment]
Only v5.6.0 doesn't have this issue, but I need to use scrollTimeReset(v5.7.0 new feature), too.
Is there any solution? Please help.

**Comment by aukolov** [comment]
Same problem... Any workaround out there?

**Comment by rloview** [comment]
@acerix @arshaw Could you please share if there is any update? I couldn't find a workaround and just waiting.

**Comment by aukolov** [comment]
Not sure if this helps anybody, but in my case the problem is gone after upgrading to `react 18` and `fullcalendar 6.1.8`, and also removing a handler to `datesSet={(arg) => {...}}` - in any case I was setting the dates from an external component using  `api.gotoDate()`. The calendar scroll is not jumping for me anymore.

**Comment by xi** [comment]
I changed the jsfiddle to use 6.1.8, but the issue remains the same: https://jsfiddle.net/b32dhn0o/

@aukolov  can you provide more details? You mentioned `datesSet` and `api.goToDate()`, but the issue is about `refetchEvents()`. How does this fit together?

**Comment by aukolov** [comment]
@xi , true, probably my scenario is different from the original one described in the ticket, just the unfortunate behavior is the same.

In my case I use `events` prop to set events, and I have an external component that manages date and view navigation. Evolutionary I ended up with code that handled a callback from this external navigation component, then updates dates and view in the calendar, then handled a callback from the calendar to save the updated dates in state, and then based on this new dates state we updated `events` prop. After I straightened up the flow by excluding the handler of dates update from the calendar and updating the state directly from the navigator component callback, the issue was gone.

**Comment by andrewbuyan** [comment]
as a workaround you can set min-height for ".fc-daygrid-body" same as height just before refetching events. then remove min-height after refetch.

**Comment by aakashglobalfpo** [comment]
I am Also Facing the issue on Polling. When the data is refetched from the API while polling, Calendar  is scrolling to the top.
